storey/steps/sample.py

```diff
@@ -0,0 +1,50 @@
+from enum import Enum
+
+from storey import Filter
+
+
+class Sample(Filter):
+    """
+    Samples stream events, passes a single event downstream accordingly to emit_policy. By default emit_policy is set to
+    EMIT_FIRST, meaning that for Sample(rate_count=n) - first event will be passed downstream, while the n-1 next events
+    are filtered out. Other wise if emit_policy is set to EMIT_LAST - the first n-1 events are filtered out, but event
```

Comment:
Otherwise

Reply:
It's not clear what happens if the stream is terminated mid-period. "Last" would indicate that the last event will be emitted, but the n-1 description would indicate that it won't be. I'm not really sure why someone might want to emit "last" anyway. It's basically the same as emitted first but skipping the first event.

Reply:
No, If i want to sample in a rate of 1 per 1000 events, then first/last are quite different events. I'll rewrite the doc.